### DISCUSSION TITLE
More models

### DISCUSSION THREAD

**Discussion by medledan** [imrayya/stable-diffusion-webui-Prompt_Generator#13] [Q&A] [answered]
Where do you go on huggingFace to get more token models to download? Not sure where to go. Thanks for your help.

**Comment by imrayya** [comment, marked as answer]
Well there no real category for it on hugginFace to get models that are specific to prompt generation, but if you search under [text-generation](https://huggingface.co/models?pipeline_tag=text-generation) and you should find a couple

Personally, I have theses ones setup to try:

- Gustavosta/MagicPrompt-Stable-Diffusion
- Gustavosta/MagicPrompt-Dalle
- FredZhang7/distilgpt2-stable-diffusion-v2
- FredZhang7/anime-anything-promptgen-v2
- AUTOMATIC/promptgen-lexart
- AUTOMATIC/promptgen-majinai-safe
- succinctly/text2image-prompt-generator
- microsoft/Promptist


They should mention which tokenizer they've used in the model cards.

Hope that helps

**Reply by medledan** [reply]
Ive gotten one working that was already in the folder; I guess another extension downloaded it. But how do I download and install the HuggingFace model? Your folders have .no_exist, blobs....this is not the same structure on HuggingFace

**Reply by imrayya** [reply]
In the json file (`.\extensions\stable-diffusion-webui-Prompt_Generator\models.json` in the Automatic1111 WebUI folder). All you need to do is input the name of the model and what tokenizer it uses and this extension should download the extension by itself (in `*username*/.cache/huggingface/hub/models`).

An example adding `microsoft/Promptist` to `models.json`

```
[
    {
        "Title":"Gustavosta",
        "Tokenizer":"gpt2",
        "Model":"Gustavosta/MagicPrompt-Dalle"
    },
    {
        "Title":"FredZhang7",
        "Tokenizer":"distilgpt2",
        "Model":"FredZhang7/distilgpt2-stable-diffusion-v2"
    },
    {
        "Title":"Microsfot Promptist",
        "Tokenizer":"gpt2",
        "Model":"microsoft/Promptist"
    }
]
```

Just by modifying the json like that, the extension should automatically download the model and use it (once you actually select it in the UI and click generate). You don't have to download anything manually

**Reply by Delteuz** [reply]
> 

Which of these models do you recommend using? and under text-generation models at hugging face, are all of them usable? How do I handpick which one of them is good?

**Reply by imrayya** [reply]
The ones I use are: 

- FredZhang7/distilgpt2-stable-diffusion-v2
- Gustavosta/MagicPrompt-Dalle
- FredZhang7/anime-anything-promptgen-v2
- FredZhang7/danbooru-tag-generator

Others that I found are (haven't tested the quality):

- microsoft/Promptist
- Gustavosta/MagicPrompt-Stable-Diffusion
- succinctly/text2image-prompt-generator
- AUTOMATIC/promptgen-lexart
- AUTOMATIC/promptgen-majinai-unsafe
- AUTOMATIC/promptgen-majinai-safe
- daspartho/prompt-extend

I haven't tested it yet but theoretically most models should work assuming you have enough ram. Including the big ones but I suggest looking for ones that uses GPT2 or DistilGPT2. 

Unfortunately, HuggingFace doesn't have the best organization to find stuff.

Personally, I mostly use FredZhang7/distilgpt2-stable-diffusion-v2 and FredZhang7/danbooru-tag-generator

**Reply by malo67** [reply]
I'm having some problems with adding models, every time I try to add a model to the json the extension tab for the prompt generator disappears in automatic1111. I'm thinking I'm just putting the info in wrong maybe?

I'm getting this error.

![Screenshot 2023-06-11 174357](https://github.com/imrayya/stable-diffusion-webui-Prompt_Generator/assets/48817283/3cdef0d9-e74e-419f-874c-9fdd38462cab)

**Reply by imrayya** [reply]
I mean I can't tell without seeing the Json file but from the error, I'm assuming you're missing a comma. 
Using my above example to demonstrate it 
````
[
    {
        "Title":"Gustavosta" <-----Missing commas between the parameters
        "Tokenizer":"gpt2",
        "Model":"Gustavosta/MagicPrompt-Dalle"
    },
    {
        "Title":"FredZhang7",
        "Tokenizer":"distilgpt2",
        "Model":"FredZhang7/distilgpt2-stable-diffusion-v2"
    } <------Missing commas between the models
    {
        "Title":"Microsfot Promptist",
        "Tokenizer":"gpt2",
        "Model":"microsoft/Promptist"
    }
]
````

If you can't find the problem, feel free to post the JSON for me to see

**Reply by malo67** [reply]
I was just missing the comma, thank you!
